### PR TITLE
chore(curriculum): prepare tests for recursion-protection

### LINKED ISSUE #36118: Client crashes with React component recursive renders
**Describe the bug**
Whenever a class component renders itself crashes the website

**To Reproduce**
1. Go to '[Default Props Challenge](https://learn.freecodecamp.org/front-end-libraries/react/override-default-props/)'
2. Type `<ShoppingCart />` instead of `<Items />`

**Expected behavior**
Should throw an error or should not render invalid code

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 
 - Browser: Chrome
 - Version: 73.0.3683.75

**Additional context**
This issue is for most of challenges

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
[skip ci]

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #36118 when https://github.com/ojeytonwilliams/loop-protect/pull/5 is merged too.

**To review**

1) Build package downloaded from my fork:

```bash
cd node_modules/@freecodecamp/loop-protect
npm ci
npm run build
```

2) Build the client as you normally would.

---

@raisedadead Should we add `ojeytonwilliams/loop-protect` to `freeCodeCamp/`?

### CODE REVIEW

File: client/src/templates/Challenges/rechallenge/transformers.js [hunk 1]
```diff
@@ -94,7 +99,7 @@ async function loadPresetReact() {
     );
   /* eslint-enable no-inline-comments */
   babelOptionsJSX = {
-    plugins: ['loopProtection'],
+    plugins: ['recursionProtection'],
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/rechallenge/transformers.js:102]: Do we not want loop protection too?

Comment (ShaunSHamilton) [reply client/src/templates/Challenges/rechallenge/transformers.js:102]: It is both now

**Comment by raisedadead** [comment]
> Should we add ojeytonwilliams/loop-protect to freeCodeCamp/?

Yes - that should be fine.

**Comment by ojeytonwilliams** [comment]
@ShaunSHamilton https://github.com/ojeytonwilliams/loop-protect/pull/5 is in.  Is it possible to get CI to work, i.e. by adding the steps in the OP to the action?

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams I suppose, yes. But that cannot be our desired change? So, can we not just publish a new version (minor), and link that here?

**Comment by ojeytonwilliams** [comment]
Kinda. The issue is that if we publish, we have to catch the renovate PR before it's merged (assuming the CI passes).

I'll have a quick look and see if there's a hack to get this working.

**Comment by ojeytonwilliams** [comment]
Also: 
image

:rocket:

**Comment by ShaunSHamilton** [comment]
@nhcarrigan I would appreciate help with the failing `learn-basic-js-by-building-a...`. Looking at the test, it it is trying to assert the Babel-ified code which does change with this.

Are you at any point rewriting this to avoid doing such?

**Comment by ShaunSHamilton** [comment]
Yes, we do know what it changes to, but this is the test I am seeing failing:
```
assert(
  pick
    .toString()
    .replace(/\s/g, '')
    .includes(
      'while(numbers.length<10){if(_LPC++%2000===0&&Date.now()-_LP>1500){'
    )
);
```

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams That largely sounds correct.

However, we need to treat this like a breaking library upgrade:
1) Deploy new version of library
2) Add these changes during upgrade

We should be able to push the test changes sans the upgrade.

**Comment by ojeytonwilliams** [comment]
> We should be able to push the test changes sans the upgrade.

I figured as much. How about we do that in this PR and let renovate handle the updates to loop-protect?

**Review by ojeytonwilliams — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**